services/QuillLMS/app/models/subscription.rb

```diff
@@ -218,16 +207,23 @@ def update_if_charge_succeeds
     end
   end
 
-  def save_if_charge_succeeds(premium_type, school=nil)
-    if premium_type === 'teacher'
-      charge = charge_user_for_teacher_premium
-      payment_amount = TEACHER_PRICE
-    elsif premium_type === 'school'
-      charge = charge_user_for_school_premium(school)
-      payment_amount = SCHOOL_RENEWAL_PRICE
+  def self.default_expiration_date(school_or_user)
+    last_subscription = school_or_user.subscriptions.active.first
+    if last_subscription.present?
+      redemption_start_date(school_or_user) + 1.year
+    elsif school_or_user.instance_of?(School)
+      promotional_dates[:expiration]
     else
-      raise "an incorrect premium type #{premium_type} was passed"
+      Date.today + 1.year
     end
+  end
+
+  def save_if_charge_succeeds(premium_type, school=nil)
```

Comment:
Looks like we're changing this method to only save school premium subs. Might want to change the method name as well to reflect that so it's not confusing.

Reply:
I will most likely be removing this method when integrating the school subscription stuff.  I decided to leave it for now so to keep the commit diff down.